lighting-estimation-explainer.md

```diff
@@ -0,0 +1,91 @@
+# WebXR Device API - Lighting Estimation
+This document explains the portion of the WebXR APIs that enable developers to render augmented reality content that reacts to real world lighting.
+
+## Introduction
+
+"Lighting Estimation" is implemented by AR platforms using a combination of sensors, cameras, algorithms, and machine learning.  Lighting estimation provides input to rendering algorithms and shaders to ensure that the shading, shadows, and reflections of objects appear natural when presented in a diverse range of settings.
+
+The XRLightProbe and XRReflectionProbe interfaces expose the values that the platform offer to WebXR rendering engines.  Their corresponding accessor functions, XRFrame.getGlobalLightEstimate() and XRFrame.getGlobalReflectionProbe() are only accessible once the first frame of an AR session has started.  The promises may be resolved on the same frame or multiple frames later, depending on the platform capabilities.  In some cases, the promises may fail, indicating that the lighting values are not available at this time and should be requested again at a later time.
+
+Although modern render engines support multiple "Light Probes" and "Reflection Probes" in a scene, the WebXR API returns only a single corresponding XRLightProbe and XRReflectionProbe, representing the global approximated lighting values to be used in the area in close proximity to the viewer.  When future platforms become capable of reporting multiple probes with precise locations away from the viewer, such support could be implemented additively without breaking changes.
+
+The orientation of the lighting information is relative to the XRViewerPose for the XRFrame that getGlobalLightEstimate() or getGlobalReflectionProbe() was requested on.  As it may be computatinonaly expensive to rotate SH and texture cubes, XRLightProbe.sphericalHarmonicsCoefficients() and XRReflectionProbe.orientation() enable the same SH and texture cubes to be used in multiple orientations.
+
+It is possible to treat a synthetic VR scene as the environment that AR content will be mixed in to.  In this case, the platform will be able to report the lighting estimation using the geometry of the VR scene.  As the WebXR API does not specifically express if the world is synthetic or real, AR content is to be written the same, without such knowledge.  Such "AR in VR" techniques do not affect the WebXR specification directly and are beyond the scope of this text.
```

Comment:
Is this proposing that we allow the UA to support this in vr mode? It wouldn't be a requirement right?

Reply:
This would be optional for UA implementations.  In this case, the session would be an immersive-ar session.  This would be driven by a UA specific UI, which requires no understanding of the "AR in VR" for the content's implementation.